File: tray.py
import gi
gi.require_version('AppIndicator3', '0.1')
from gi.repository import AppIndicator3 as app

# 以下是继续实现托盘的步骤：

APPINDICATOR_ID = 'myappindicator'

def main():
    # 创建托盘图标
    indicator = app.Indicator.new(
        APPINDICATOR_ID,
        "icon",
        app.IndicatorCategory.APPLICATION_STATUS
    )
    indicator.set_status(app.IndicatorStatus.ACTIVE)
    indicator.set_attention_icon("icon")

    # 显示或隐藏托盘图标
    if indicator.get_status() == app.IndicatorStatus.ACTIVE:
        indicator.set_status(app.IndicatorStatus.PASSIVE)
    else:
        indicator.set_status(app.IndicatorStatus.ACTIVE)

    # 删除托盘图标
    indicator.set_status(app.IndicatorStatus.PASSIVE)
    app.Indicator.uninstall(indicator)

File: tray.cpp
#include <libappindicator/app-indicator.h>

#define APPINDICATOR_ID "myappindicator"

int main() {
  // 创建托盘图标
  AppIndicator *indicator = app_indicator_new(APPINDICATOR_ID, "icon", APP_INDICATOR_CATEGORY_APPLICATION_STATUS);
  app_indicator_set_status(indicator, APP_INDICATOR_STATUS_ACTIVE);
  app_indicator_set_attention_icon(indicator, "icon");

  // 显示或隐藏托盘图标
  if (app_indicator_get_status(indicator) == APP_INDICATOR_STATUS_ACTIVE) {
    app_indicator_set_status(indicator, APP_INDICATOR_STATUS_PASSIVE);
  } else {
    app_indicator_set_status(indicator, APP_INDICATOR_STATUS_ACTIVE);
  }

  // 删除托盘图标
  app_indicator_set_status(indicator, APP_INDICATOR_STATUS_PASSIVE);
  app_indicator_stop(indicator);
  return 0;
}